query/src/main/java/org/infinispan/query/affinity/AffinityIndexManager.java

```diff
@@ -155,26 +179,23 @@ private void handleOwnershipAcquired() {
 
    @Override
    public void flushAndReleaseResources() {
-      Address lockHolder = getLockHolder(getIndexName());
-      if (lockHolder != null) {
-         InfinispanDirectoryProvider directoryProvider = (InfinispanDirectoryProvider) getDirectoryProvider();
-         int activeDeleteTasks = directoryProvider.getActiveDeleteTasks();
-         boolean wasInterrupted = false;
-         while (activeDeleteTasks > 0) {
-            try {
-               Thread.sleep(POLL_WAIT);
-               logDebug("Waiting for pending delete tasks, remaining: %s", activeDeleteTasks);
-               activeDeleteTasks = directoryProvider.getActiveDeleteTasks();
-            } catch (InterruptedException e) {
-               wasInterrupted = true;
-            }
-         }
-         if (wasInterrupted) {
-            Thread.currentThread().interrupt();
+      InfinispanDirectoryProvider directoryProvider = (InfinispanDirectoryProvider) getDirectoryProvider();
+      int activeDeleteTasks = directoryProvider.pendingDeleteTasks();
+      boolean wasInterrupted = false;
+      while (activeDeleteTasks > 0) {
+         try {
+            Thread.sleep(POLL_WAIT);
```

Comment:
I don't think those delete tasks will deadlock (or at least was not able to reproduce it under load), they are a bunch of ```cache.remove``` being done by a separate executor.
I will cap that waiting time until replacing this executor with something better.